File: 72-edit-distance/edit-distance.java
class Solution {

    public int minDistance(String word1, String word2) {

        int n = word1.length();
        int m = word2.length();

        int[][] dp = new int[n + 1][m + 1];

        for (int i = 0; i <= n; i++) {
            dp[i][m] = n - i;
        }

        for (int j = 0; j <= m; j++) {
            dp[n][j] = m - j;
        }

        for (int i = n - 1; i >= 0; i--) {

            for (int j = m - 1; j >= 0; j--) {
                if(word1.charAt(i)==word2.charAt(j)){
                    dp[i][j]=dp[i+1][j+1];
                }
                else{
                    int insert=1+dp[i][j+1];
                    int delete=1+dp[i+1][j];
                    int replace=1+dp[i+1][j+1];
                    dp[i][j]=Math.min(insert,Math.min(delete,replace));
                }

            }
        }

        return dp[0][0];
    }
}

File: leet-sync/72-edit-distance/edit-distance.py
class Solution(object):
    def minDistance(self, word1, word2):
        dp = [[0]* (len(word1)+1) for k in range(len(word2)+1)] 
        for i in range(len(word1)):
            dp[len(word2)][i]=len(word1)-i
        for j in range(len(word2)):
            dp[j][len(word1)]=len(word2)-j
        for i in range(len(word2)-1,-1,-1):
            for j in range(len(word1)-1,-1,-1):
                if(word1[j]==word2[i]):
                    dp[i][j]=dp[i+1][j+1]
                else:
                    insert=dp[i+1][j]
                    delete=dp[i][j+1]
                    replace=dp[i+1][j+1]
                    dp[i][j]=1+min(insert,delete,replace)
        return dp[0][0]